src/instructions/interrupts.rs

```diff
@@ -152,19 +152,14 @@ pub fn int3() {
 /// This currently needs to be a macro because the `int` argument needs to be an
 /// immediate. This macro will be replaced by a generic function when support for
 /// const generics is implemented in Rust.
-#[cfg(feature = "inline_asm")]
+#[cfg_attr(docsrs, doc(cfg(feature = "nightly")))]
 #[macro_export]
 macro_rules! software_interrupt {
     ($x:expr) => {{
+        #[cfg(feature = "inline_asm")]
         asm!("int {id}", id = const $x, options(nomem, nostack));
-    }};
-}
 
-/// Not implemented
-#[cfg(not(feature = "inline_asm"))]
-#[macro_export]
-macro_rules! software_interrupt {
-    ($x:expr) => {{
-        compile_error!("software_interrupt not implemented for non-nightly");
+        #[cfg(not(feature = "inline_asm"))]
+        compile_error!("software_interrupt!() requires Nightly Rust");
```

Comment:
We should probably mention that the `nightly` feature needs to be enabled (using a nightly compiler is not enough).